src/Bridges/Constraint/bridges/split_hyperrectangle.jl

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    SplitHyperRectangleBridge{T,G,F} <: Bridges.Constraint.AbstractBridge
+
+`SplitHyperRectangleBridge` implements the following reformulation:
+
+  * ``f(x) \\in \\textsf{HyperRectangle}(l, u)`` to
+    ``[f(x) - l; u - f(x)] \\in \\mathbb{R}_+``.
+
+## Source node
+
+`SplitHyperRectangleBridge` supports:
+
+  * `F` in [`MOI.HyperRectangle`](@ref)
+
+## Target nodes
+
+`SplitHyperRectangleBridge` creates:
+
+  * `G` in [`MOI.Nonnegatives`](@ref)
+"""
+mutable struct SplitHyperRectangleBridge{T,G,F} <: AbstractBridge
+    ci::MOI.ConstraintIndex{G,MOI.Nonnegatives}
+    func::F
+    set::MOI.HyperRectangle{T}
+end
+
+const SplitHyperRectangle{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{SplitHyperRectangleBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{SplitHyperRectangleBridge{T,G,F}},
+    model::MOI.ModelLike,
+    f::F,
+    s::MOI.HyperRectangle,
+) where {T,G,F}
+    scalars = MOI.Utilities.eachscalar(f)
+    lower = [
+        MOI.Utilities.operate(-, T, fi, s.lower[i]) for
+        (i, fi) in enumerate(scalars) if isfinite(s.lower[i])
+    ]
+    upper = [
+        MOI.Utilities.operate(-, T, s.upper[i], fi) for
+        (i, fi) in enumerate(scalars) if isfinite(s.upper[i])
+    ]
+    g = MOI.Utilities.operate(vcat, T, lower..., upper...)
+    ci = MOI.add_constraint(model, g, MOI.Nonnegatives(MOI.output_dimension(g)))
+    return SplitHyperRectangleBridge{T,typeof(g),F}(ci, f, s)
+end
+
+function MOI.supports_constraint(
+    ::Type{<:SplitHyperRectangleBridge{T}},
+    ::Type{<:MOI.AbstractVectorFunction},
+    ::Type{MOI.HyperRectangle{T}},
+) where {T}
+    return true
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:SplitHyperRectangleBridge},
+)
+    return Tuple{Type}[]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{<:SplitHyperRectangleBridge{T,G}},
+) where {T,G}
+    return Tuple{Type,Type}[(G, MOI.Nonnegatives),]
+end
+
+function concrete_bridge_type(
+    ::Type{<:SplitHyperRectangleBridge{T}},
+    ::Type{F},
+    ::Type{MOI.HyperRectangle{T}},
+) where {T,F<:MOI.AbstractVectorFunction}
+    G = MOI.Utilities.promote_operation(-, T, F, Vector{T})
+    return SplitHyperRectangleBridge{T,G,F}
+end
+
+function MOI.get(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    bridge::SplitHyperRectangleBridge,
+)
+    return bridge.func
```

Comment:
@blegat any ideas for how to recover the function in the general case? Like if we have a `-Inf, Inf` row, then we'll drop the row from the bridge, so we at least need to store those functions, and if someone passes `f(x) + c in HyperRectangle(l, u)`, then we need to store `l` and `u` to be able to recover non-zero constants `c`.

Reply:
We could error on nonzero constant like for the scalar affine sets. With infinite bounds, we could store the function yes

Reply:
This is probably okay for now. We can see if anyone complains? I don't think it's a common set.

Reply:
We should add a TODO or FIXME with a link to the docs where we say why we don't want that then it's fine to merge

Reply:
Done.